### PR TITLE
Enable asan.test_embind_val and add asan.embind* to github ci

### PR DESCRIPTION
**Pull request by kripken**

### CODE REVIEW

**Comment by RReverser** [comment]
I'm surprised the CI doesn't trigger and fail on leak checks from https://github.com/emscripten-core/emscripten/issues/11533. Doesn't ASAN automatically include/enable LSAN as well?

File: .circleci/config.yml [hunk 1]
```diff
@@ -503,7 +503,7 @@ jobs:
     executor: bionic
     steps:
       - run-tests:
-          test_targets: "wasm2"
+          test_targets: "wasm2 asan.test_embind*"
```
Comment (sbc100) [inline .circleci/config.yml:506]: Do we want to rename this job ?  Or create a new asan job?

Comment (kripken) [reply .circleci/config.yml:506]: It's just a handful of tests, so a new job seems excessive.

Renaming seems possible, but I don't have a great idea...?

Comment (sbc100) [reply .circleci/config.yml:506]: Can't think of a good one :(

**Comment by kripken** [comment]
> Doesn't ASAN automatically include/enable LSAN as well?

Not sure. I seem to recall there is some amount that is, but maybe not all of it?

**Comment by sbc100** [comment]
I guess we don't enable it by default?    https://clang.llvm.org/docs/AddressSanitizer.html#memory-leak-detection  ??

**Comment by RReverser** [comment]
Uhh that's at least not what our docs say https://emscripten.org/docs/debugging/Sanitizers.html, and they even have examples where `-fsanitize=address` finds leaks as well...

**Comment by RReverser** [comment]
Looks like at least in `asani` mode we explicit disable leaks:

https://github.com/emscripten-core/emscripten/blob/121ca5561eec983848c8cfb0538867fb86734af3/tests/test_core.py#L8908

https://github.com/emscripten-core/emscripten/blob/121ca5561eec983848c8cfb0538867fb86734af3/tests/asan-no-leak.js#L1

But that doesn't seem to be the mode used here, so they should work?

**Comment by kripken** [comment]
Landing. Let's open an issue for the separate leak topic if we think that needs to be investigated.

**Comment by RReverser** [comment]
It just seems suspicious because makes me wonder whether other ASAN options run here.

**Comment by sbc100** [comment]
Oh, maybe because that test (like almost all our tests) doesn't use `EXIT_RUNTIME=1`.  I imagine leak checking depends on that setting.   

We should probably use `EXIT_RUNTIME=1` in a lot more tests...  but there is also a rather nasty bug waiting for us when we enable it: https://bugs.llvm.org/show_bug.cgi?id=45876

**Comment by RReverser** [comment]
Ah, if `EXIT_RUNTIME` is disabled (I believe it's enabled by default?) then it makes sense indeed.

**Comment by RReverser** [comment]
Perhaps worth enabling it for most tests and then selectively exclude those that trigger the bug? (it's not clear to me how easy / frequent it is)

**Comment by sbc100** [comment]
`NO_EXIT_RUNTIME` is the default.  Most emscripten users are tend to build "reactors" rather then "commands".   It pretty rare that anyone actually cares about the running of atexit handlers and its very common that someone tries to call a function from JS after main has returned.   Its also a minor code size saving since all the exiting handling code can get DCEd.

There are a good number of tests that don't want exit runtime set (because they call things after main has run) so we probably want to go through an annotate the tests in some way.   Since it only really matters for lsan I'm not too worried about getting right for every single test as long as we have reasonable lsan coverage in there.

**Comment by tlively** [comment]
Yeah we have some possibly-counterintuitive logic that `-fsanitize=leak` also enables the exit runtime so that leaks will be checked but `-fsanitize=address` does not. This should probably either be changed or documented, but I'm not sure what behavior is best for users.

**Pull request merged**